### PR TITLE
feat: trace http.status_code

### PR DESCRIPTION
**Pull request by xiang-xx**
Trace `http.status_code`

Add a `traceResponseWriter` to set trace span attribute and status:
- set attribute `http.status_code` with http statusCode
- set `Error` when `code >= 400`, else `Ok`

### CODE REVIEW

File: rest/handler/tracinghandler.go [hunk 1]
```diff
@@ -7,18 +7,42 @@ import (
...
+)
+
 var notTracingSpans sync.Map
 
 // DontTraceSpan disable tracing for the specified span name.
 func DontTraceSpan(spanName string) {
 	notTracingSpans.Store(spanName, lang.Placeholder)
 }
 
+type traceResponseWriter struct {
```
Comment (kevwan) [inline rest/handler/tracinghandler.go:28]: How is the logic with Hijack/Flush method?

Please check it out in `loghandler.go`.

File: rest/handler/tracinghandler.go [hunk 2]
```diff
@@ -48,7 +72,21 @@ func TracingHandler(serviceName, path string) func(http.Handler) http.Handler {
...
+				code: http.StatusOK,
+			}
+			next.ServeHTTP(trw, r.WithContext(spanCtx))
+
+			span.SetAttributes(attribute.KeyValue{
+				Key:   traceKeyStatusCode,
+				Value: attribute.IntValue(trw.code),
+			})
+			if trw.code >= http.StatusInternalServerError {
+				span.SetStatus(codes.Error, http.StatusText(trw.code))
```
Comment (kevwan) [inline rest/handler/tracinghandler.go:86]: Why only keep codes >= 500?

**Review by kevwan — CHANGES_REQUESTED** [review]
Thanks for you PR!

File: rest/handler/tracinghandler.go (hunk 1 above)
Comment (xiang-xx) [reply rest/handler/tracinghandler.go:31]: Added. Thanks for your review.

File: rest/handler/tracinghandler.go (hunk 2 above)
Comment (xiang-xx) [reply rest/handler/tracinghandler.go:105]: Update to >= 400, 4xx is also an abnormal condition and requires a warning.

**Review by kevwan — APPROVED** [review]
LGTM.

File: rest/handler/tracinghandler.go [hunk 3]
```diff
@@ -48,7 +91,21 @@ func TracingHandler(serviceName, path string) func(http.Handler) http.Handler {
...
+			next.ServeHTTP(trw, r.WithContext(spanCtx))
+
+			span.SetAttributes(attribute.KeyValue{
+				Key:   traceKeyStatusCode,
+				Value: attribute.IntValue(trw.code),
+			})
+			if trw.code >= http.StatusBadRequest {
+				span.SetStatus(codes.Error, http.StatusText(trw.code))
+			} else {
+				span.SetStatus(codes.Ok, "")
```
Comment (kevwan) [inline rest/handler/tracinghandler.go:107]: Is `span.SetStatus(codes.Ok, http.StatusText(trw.code))` better?

Comment (xiang-xx) [reply rest/handler/tracinghandler.go]: Yes, fixed.

**Pull request merged**